src/app/code/community/Omise/Gateway/Model/Payment/Offsitepaynow.php

```diff
@@ -0,0 +1,162 @@
+<?php
+class Omise_Gateway_Model_Payment_Offsitepaynow extends Omise_Gateway_Model_Payment_SimpleOffsite_Payment
+{
+
+    /**
+     * @var string
+     */
+    protected $_code = 'omise_offsite_paynow';
+
+    /**
+     * @var string
+     */
+    protected $_formBlockType = 'omise_gateway/form_default';
+
+    /**
+     * @var string
+     */
+    protected $_infoBlockType = 'payment/info';
+
+    /**
+     * @var string
+     */
+    protected $_callbackUrl = 'omise/callback_validateoffsitepaynow';
+
+    /**
+     * @var array
+     */
+    protected $_currencies = array('SGD');
+
+    /**
+     * @var string
+     */
+    protected $_type = 'paynow';
+
+    /**
+     * Payment Method features
+     *
+     * @var bool
+     */
+    protected $_isGateway          = true;
+    protected $_canReviewPayment   = true;
+    protected $_isInitializeNeeded = true;
+    protected $_allowRedirect      = false;
+
+    /**
+     * Instantiate state and set it to state object
+     *
+     * @param string        $payment_action
+     * @param Varien_Object $state_object
+     */
+    public function initialize($payment_action, $state_object) {
```

Comment:
This function is identical to Tesco payment method. We should probably add a parent class for QR/Barcode methods like these as they seem to share a bunch of stuff

Reply:
Which one do you prefer, moving code to new function in parent class OR complete new parent class extended by this class?

Reply:
It seems to me like these barcode type methods probably should be subclassed from a barcode parent class - as the current parent is also used by other methods, right? These barcode type methods are like another family of offsite methods

Reply:
Okay, Good Idea!